pandas/formats/format.py

```diff
@@ -1575,13 +1586,22 @@ def _save_chunk(self, start_i, end_i):
                 self.data[col_loc] = col
 
         ix = data_index.to_native_types(slicer=slicer, na_rep=self.na_rep,
+                                        bytes_encoding=self.bytes_encoding,
                                         float_format=self.float_format,
                                         decimal=self.decimal,
                                         date_format=self.date_format,
                                         quoting=self.quoting)
 
         lib.write_csv_rows(self.data, ix, self.nlevels, self.cols, self.writer)
 
+    def _bytes_to_str(self, values):
+        """Modify values list by decoding bytes to str."""
```

Comment:
so I think if we have mixed bytes & strings then we should simply raise a ValueError.

you can use ``lib.is_bytes_array(arr)``, pass it an array (e.g. here you could do ``np.asarray(values)`` to ensure the arr)